packages/edge/ts/models/ExperienceEvent.ts

```diff
@@ -1,29 +1,49 @@
-/*
-Copyright 2022 Adobe. All rights reserved.
-This file is licensed to you under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License. You may obtain a copy
-of the License at http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software distributed under
-the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
-OF ANY KIND, either express or implied. See the License for the specific language
-governing permissions and limitations under the License.
-*/
+type ExperienceEventArguments =
+  | {
+      xdmData: Record<string, any>;
+      data?: Record<string, any> | null;
+      datasetIdentifier?: string | null;
+    }
+  | {
+      xdmData: Record<string, any>;
+      data?: Record<string, any> | null;
+      datastreamIdOverride?: string | null;
+      datastreamConfigOverride?: Record<string, any> | null;
+    };
 
 class ExperienceEvent {
-  xdmData?: Record<string, any>;
+  xdmData: Record<string, any>;
```

Comment:
Now that you are changing this from optional to required, seems like you can still pass in an empty map for `xdmData` but you need to specify it as param?

Reply:
Good call, changed it back to optional. empty or null xdmData would create an empty ExperienceEvent and there was a log to provide the warning already.
Also logic is covered in the native platform.